### PR TITLE
[feat] Create application backend

### PR DESCRIPTION
**Pull request by plaanty**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Created the query request for an application based on Inaya's new table in supabase
- Linked this to the new application button found on the main page dashboard

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- Make sure that the link to the pre-app is accurate?

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Further integrate it such that it can actually store the application data in the supabse table when the applicant makes it

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

File: assets/images/coolicon.png
Comment (jinkang-0) [inline assets/images/coolicon.png]: We should prefer to use SVG custom icons or pull from React Icons if possible. PNGs don't scale well in the web

File: assets/images/Log_Out.png
Comment (jinkang-0) [inline assets/images/Log_Out.png]: We should prefer to use SVG custom icons or pull from React Icons if possible. PNGs don't scale well in the web

File: lib/supabase/application.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  // Add all the columns from adopter_applications_dummy table
+  app_uuid: UUID;
+  adopter_uuid: UUID;
+  personal_bio: string;
+  gender_pref: string;
+  return_explanation: string;
+  in_complete: boolean;
+  accepted: boolean;
+  rejected: boolean;
+};
```
Comment (jinkang-0) [inline lib/supabase/application.ts]: After rebasing, you should be able to extract the type from the Supabase generated types. Here are some general steps:
1. Define an `AdopterApplication` type in `schema.ts` to be `Database['public']['Tables']['adopter_applications_dummy']['Row']`
2. Import the type here to use

File: lib/supabase/application.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export async function createAdopterApplication(
+  applicationData: Omit<AdopterApplication, 'id' | 'created_at' | 'updated_at'>,
```
Comment (jinkang-0) [inline lib/supabase/application.ts]: Is there a reason why we are omitting these fields? If the rationale is to set those fields to null, I'm sorry to report that Typescript does not modify those fields just because you omitted the type requirement. If we pass in a full `AdopterApplication` object (which includes the omitted fields), the object received will still fully retain those fields, and it _will_ be passed to the Supabase call.

If we want to ensure the `id`, `created_at`, and `updated_at` fields are null, we should set them explicitly.

File: lib/supabase/application.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  const supabase = getSupabaseBrowserClient();
+
+  const { data, error } = await supabase
+    .from('adopter_applications_dummy')
+    .insert([applicationData])
+    .select()
+    .single();
+
+  if (error) {
+    console.error('Error creating application:', error);
```
Comment (jinkang-0) [inline lib/supabase/application.ts]: nit: Prefer `Logger.error` rather than `console.error`.

**Review by jinkang-0 — CHANGES_REQUESTED** [review]
Please rebase this PR off the style main app PR, after you rebase that one from main

File: lib/supabase/application.ts (hunk 2 above)
Comment (plaanty) [reply lib/supabase/application.ts]: Honestly, I'm not sure why they're set to be omitted. I'm just going to delete that line.

**Comment by ethan-tam33** [comment]
@plaanty I think you rebased this incorrectly, could you fix this asap?

**Comment by plaanty** [comment]
> @plaanty I think you rebased this incorrectly, could you fix this asap?

Ok, I can rebase from main again and see if that fixes it.

**Comment by plaanty** [comment]
> it looks like you pulled instead of rebased, resulting in lots of old commits... do you want to work on this during final hack?

@jinkang-0 what I can do before final hack is go back to an old commit prior to rebasing, then rebase, then manually copy my work over to that older commit, then force push to get rid of the bad rebasing
I think this should work?
